overlord/snapstate/handlers.go

```diff
@@ -401,6 +403,17 @@ func installInfoUnlocked(st *state.State, snapsup *SnapSetup) (*snap.Info, error
 	return installInfo(st, snapsup.InstanceName(), snapsup.Channel, snapsup.Revision(), snapsup.UserID)
 }
 
+// autoRefreshRateLimited returns the rate limit of auto-refreshes or 0 if
+// there is no limit.
+func autoRefreshRateLimited(st *state.State) (rate float64) {
+	tr := config.NewTransaction(st)
+	// FIXME: support user friendly values like "5M"
```

Comment:
Indeed. Let's please not merge this without this being enforced, so we never have to interpret naked numbers assuming some guessed unit.

Alternatively, we might come up with a nice name that includes the unit instead of calling it rate-limit. I can't come up with a good one because I'm not sure about how much people would want to rate-limit.. that would make different units practical. Needs some thinking.

Reply:
I added code for this now, I split it out in https://github.com/snapcore/snapd/pull/5719 for easier review.